src/System.Management.Automation/engine/CommandCompletion/CompletionHelpers.cs

```diff
@@ -140,5 +138,40 @@ internal static bool CompletionRequiresQuotes(string completion)
 
         private static bool ContainsCharsToCheck(ReadOnlySpan<char> text)
             => text.ContainsAny(s_defaultCharsToCheck);
+
+        /// <summary>
+        /// Quotes and escapes a given completion text based on the specified options.
+        /// </summary>
+        /// <param name="completionText">
+        /// The text to be quoted and potentially escaped.
+        /// </param>
+        /// <param name="quote">
+        /// The quote character to use for enclosing the text. Defaults to a single quote if not provided.
+        /// </param>
+        /// <param name="escapeSingleQuoteChars">
+        /// Indicates whether single quote characters in the text should be escaped. Defaults to <c>true</c>.
+        /// </param>
+        /// <returns>
+        /// The quoted and optionally escaped version of the <paramref name="completionText"/>.
+        /// </returns>
+        internal static string QuoteCompletionText(
+            string completionText,
+            string quote,
+            bool escapeSingleQuoteChars = true)
```

Comment:
If we don't use the parameter why do we need it? I think escaping rules should be the same always. I suggest to remove.

Reply:
Removed `escapeSingleQuoteChars` parameter in c8c8404dfcea651f008a22dce99c594d4baec3fd